### PR TITLE
3 contributor class structure

### LINKED ISSUE #3: Feature: Add ContributorClass structure
## Status Quo
Currently there is no way to have a specific contributor classes with personalized characteristics.

## Acceptance criteria
- Have personalized cap for Class of contributors
- Have general cap for class of contributors
- Have specific seed price for class of contributors
- Have specific vesting duration for the class
- Have a  possibility to create a contributor class

### PR DESCRIPTION
**Pull request by BritikovKI**
Closes #3

### CODE REVIEW

**Review by arseneeth — CHANGES_REQUESTED** [review]
changes requested

File: contracts/seed/Seed.sol [hunk 1]
```diff
@@ -65,6 +65,8 @@ contract Seed {
     uint256 public fundingCollected; // Amount of funding tokens collected by the seed contract.
     uint256 public fundingWithdrawn; // Amount of funding token withdrawn from the seed contract.
 
+    ContributorClass[] classes; // Different contributor classes
```
[the remark this answers is not in the document]
Comment (BritikovKI) [reply contracts/seed/Seed.sol]: Fixed

File: contracts/seed/Seed.sol [hunk 2]
```diff
@@ -79,10 +81,19 @@ contract Seed {
...
 
     struct FunderPortfolio {
+        uint8   class; // Contibutor class id
         uint256 totalClaimed; // Total amount of seed tokens claimed
         uint256 fundingAmount; // Total amount of funding tokens contributed
     }
 
+    struct ContributorClass {
+        uint256 classCap; // Amount of tokens that can be donated for class
+        uint256 cap; // Amount of tokens that can be donated by specific contributor
```
[the remark this answers is not in the document]
Comment (BritikovKI) [reply contracts/seed/Seed.sol]: Fixed

**Review by arseneeth — APPROVED** [review]
lgtm

**Pull request merged**